ast.cpp

```diff
@@ -32,31 +32,76 @@ class AstNode {
 class StmtNode : public AstNode {};
 /// @note This is an abstract class.
 class ExprNode : public AstNode {};
+/// @note This is an abstract class.
+class DeclNode : public AstNode {
+ public:
+  DeclNode(const std::string& id, std::unique_ptr<ExprNode> init = {})
+      : id_{id}, init_{std::move(init)} {}
+
+  void CodeGen() const override {}
+  void Dump(int pad) const override {
+    std::cout << Pad(pad) << '(' << id_;
+    if (init_) {
+      std::cout << " = " << std::endl;
```

Comment:
Trailing white space can be unsightly and could impact the clarity of `git diff` :disappointed:
Let's revert it to `" ="` :grinning: